src/components/ArticleList.js

```diff
@@ -30,6 +48,34 @@ const ArticleList = ({ topic }) => {
     [ topic, isLoading ]
   );
 
+  const handleChange = (e) => {
+    if (!topic) {
+      getArticlesSortNoTopic(e.target.value).then((sortedArticles) => {
+        setArticles(sortedArticles.articles);
+        setSort(e.target.value);
+      });
+    } else {
+      getArticlesSort(topic, e.target.value).then((sortedArticles) => {
+        setArticles(sortedArticles.articles);
+        setSort(e.target.value);
+      });
+    }
+  };
+
+  const handleOrder = (e) => {
+    if (!order) {
+      getArticlesOrderNoTopic(e.target.value).then((orderedArticles) => {
+        setArticles(orderedArticles.articles);
+        setOrder(e.target.value);
+      });
+    } else {
+      getArticlesOrder(topic, e.target.value).then((orderedArticles) => {
+        setArticles(orderedArticles.articles);
+        setOrder(e.target.value);
+      });
+    }
+  };
+
```

Comment:
There is a lot of logic here to handle requests in different ways - if we rely on the useEffect to make the calls we can make this DRYer. You can probably do all of this with a single api function rather than multiple functions that essentially handle the same thing but in slightly different circumstances. Better to handle this logic in the api function.